addons/stock/models/product.py

```diff
@@ -175,8 +175,8 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
             # Calculate the moves that were done before now to calculate back in time (as most questions will be recent ones)
             domain_move_in_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_in_done
             domain_move_out_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_out_done
-            moves_in_res_past = dict((item['product_id'][0], item['product_qty']) for item in Move._read_group(domain_move_in_done, ['product_id', 'product_qty'], ['product_id'], orderby='id'))
-            moves_out_res_past = dict((item['product_id'][0], item['product_qty']) for item in Move._read_group(domain_move_out_done, ['product_id', 'product_qty'], ['product_id'], orderby='id'))
+            moves_in_res_past = dict((item['product_id'][0], item['product_qty']) for item in Move._read_group(domain_move_in_done, ['product_id', 'quantity_done'], ['product_id'], orderby='id'))
+            moves_out_res_past = dict((item['product_id'][0], item['product_qty']) for item in Move._read_group(domain_move_out_done, ['product_id', 'quantity_done'], ['product_id'], orderby='id'))
```

Comment:
```suggestion
            moves_in_res_past = dict((item['product_id'][0], item['quantity_done']) for item in Move._read_group(domain_move_in_done, ['product_id', 'quantity_done'], ['product_id'], orderby='id'))
            moves_out_res_past = dict((item['product_id'][0], item['quantity_done']) for item in Move._read_group(domain_move_out_done, ['product_id', 'quantity_done'], ['product_id'], orderby='id'))
```
it is really tested?

Reply:
Hi @MiquelRForgeFlow 
from commit : https://github.com/odoo/odoo/pull/130342 `product_uom_qty` will always indicate the demand qty. So, calculating inventory quantity at a point in the past based on product_qty will be inaccurate.